### PR TITLE
Use hosts-replaced URL only when loading resources

### PR DESCRIPTION
**Pull request by jgraham**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6416)

### CODE REVIEW

File: components/script/Cargo.toml [hunk 1]
```diff
@@ -21,6 +21,9 @@ path = "../util"
 [dependencies.msg]
 path = "../msg"
 
+[dependencies.net]
+path = "../net"
```
Comment (jdm) [inline components/script/Cargo.toml]: Nope! net_traits exists for a reason.

Comment (jgraham) [inline components/script/Cargo.toml]: Oh look, a @jdm!

You want the whole of the HOSTS_FILE stuff moved into net_traits then?

Comment (Manishearth) [inline components/script/Cargo.toml]: Yes. Or make it polymorphic on a trait in `net_traits`.  The former is okay.

**Comment by Manishearth** [comment]
Mostly LGTM except for the dependency.

Also, there should be some comments about what the various `_url` variables mean.

**Comment by Ms2ger** [comment]
```
components/compositing/constellation.rs:602: (much) overlong line
components/compositing/constellation.rs:605: (much) overlong line
```

File: components/net_traits/Cargo.toml [hunk 2]
```diff
@@ -24,4 +24,5 @@ log = "*"
 url = "0.2.35"
 hyper = "0.5"
 euclid = "0.1"
-
+regex = "0.1.33"
+regex_macros = "0.1.19"
```
Comment (waddlesplash) [inline components/net_traits/Cargo.toml]: Any reason you got rid of the EOL at the end of the file?

File: components/net_traits/lib.rs [hunk 3]
```diff
@@ -1,17 +1,21 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
```
Comment (waddlesplash) [inline components/net_traits/lib.rs]: Isn't there supposed to be a newline between the license header and the first `#!` line?

**Comment by jdm** [comment]
```
   Compiling servo v0.0.1 (file:///home/servo/buildbot/slave/gonk/build/ports/gonk)
src/main.rs:67:5: 67:31 error: unresolved name `resource_task::global_init`
src/main.rs:67     resource_task::global_init();
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to previous error
Could not compile `b2s`.
```

**Comment by jdm** [comment]
@bors-servo: r+

Geeze, I didn't realize this was just waiting for review of a one line change to fix a build error :(

**Comment by jdm** [comment]
```
   Compiling net_tests v0.0.1 (file:///home/servo/buildbot/slave/linux1/build/components/servo)
/home/servo/buildbot/slave/linux1/build/tests/unit/net/resource_task.rs:5:45: 5:60 error: unresolved import `net::resource_task::parse_hostsfile`. There is no `parse_hostsfile` in `net::resource_task`
/home/servo/buildbot/slave/linux1/build/tests/unit/net/resource_task.rs:5 use net::resource_task::{new_resource_task, parse_hostsfile, replace_hosts};
                                                                                                                      ^~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/net/resource_task.rs:5:62: 5:75 error: unresolved import `net::resource_task::replace_hosts`. There is no `replace_hosts` in `net::resource_task`
/home/servo/buildbot/slave/linux1/build/tests/unit/net/resource_task.rs:5 use net::resource_task::{new_resource_task, parse_hostsfile, replace_hosts};
                                         
```

**Comment by metajack** [comment]
@jgraham ping

**Comment by jdm** [comment]
Ack. One more rebase required, sorry. Please leave a comment so we notice and can r+ it straight away.

**Comment by jgraham** [comment]
Pushed another rebase.

**Comment by jdm** [comment]
```
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6:45: 6:60 error: unresolved import `net::resource_task::parse_hostsfile`. There is no `parse_hostsfile` in `net::resource_task` [E0432]
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6 use net::resource_task::{new_resource_task, parse_hostsfile, replace_hosts};
                                                                                                                     ^~~~~~~~~~~~~~~
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6:45: 6:60 help: run `rustc --explain E0432` to see a detailed explanation
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6:62: 6:75 error: unresolved import `net::resource_task::replace_hosts`. There is no `replace_hosts` in `net::resource_task` [E0432]
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6 use net::resource_task::{new_resource_task, parse_hostsfile, replace_hosts};
                                                                                                                                      ^~~~~~~~~~~~~
/Users/servo/buildbot/slave/mac2/build/tests/unit/net/resource_task.rs:6:62: 6:75 help: run `rustc --explain E0432` to see a detailed explanation
error: aborting due to 2 previous errors
Could not compile `net_tests`.
```
Womp womp.

**Comment by jdm** [comment]
@bors-servo: r+
Makes sense.

**Comment by jdm** [comment]
```


/XMLHttpRequest/timeout-cors-async.htm
--------------------------------------
FAIL XMLHttpRequest: timeout event and cross-origin request
```

**Comment by jdm** [comment]
This PR really doesn't want to merge!

**Comment by jgraham** [comment]
Seems like there shouldn't have been any metadata updates at all. Which makes some sense as wptrunner hasn't been updated to use the right host name.

**Pull request merged**